### PR TITLE
feat(AI Transform Node): UX improvements

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

- If the Code is empty or if I edited the prompt and I click on “Test step”, generate the code first
- If you edit the prompt and forget to click ‘generate code’ before running the node, show a warning
- “Transformation Code” -> “Generated JavaScript”
- “Reset value” -> “Clear contents”
-  tooltip for the code param -> hint at the bottom of the code box instead

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1864/improve-ux-in-ai-transform-node

### CODE REVIEW

File: packages/editor-ui/src/components/ButtonParameter/utils.ts [hunk 1]
```diff
@@ -44,3 +51,41 @@ export function getSchemas() {
 		parentNodesSchemas,
 	};
 }
+
+export async function generaCodeForAiTransform(prompt: string, path: string) {
```
Comment (ShireenMissi) [inline packages/editor-ui/src/components/ButtonParameter/utils.ts:55]:
```suggestion
export async function generateCodeForAiTransform(prompt: string, path: string) {
```

File: packages/editor-ui/src/components/NodeExecuteButton.vue [hunk 2]
```diff
@@ -227,6 +240,39 @@ function onMouseOver() {
 }
 
 async function onClick() {
+	if (
+		node.value?.type === AI_TRANSFORM_NODE_TYPE &&
+		!node.value?.parameters?.jsCode &&
+		node.value?.parameters?.instructions
+	) {
+		// Generate code if user didn't clicked 'Generate Code' button
```
Comment (ShireenMissi) [inline packages/editor-ui/src/components/NodeExecuteButton.vue:248]:
```suggestion
		// Generate code if user hasn't clicked 'Generate Code' button
```

**Review by ShireenMissi — COMMENTED** [review]
Tested Manually all words as expected 🎉
left some minor spelling comments
also a test is failing

**Review by ShireenMissi — APPROVED** [review]
LGTM 🎉

**Comment by janober** [comment]
Got released with [`n8n@1.67.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.67.0)

**Pull request merged**